### DISCUSSION TITLE
Auto Segmentation of 2D Images -- Fail

### DISCUSSION THREAD

**Discussion by gorin-research-repo** [Project-MONAI/MONAILabel#1247] [Q&A]
I am using MonaiLabel and Slicer 3D

I have a project with 2D images, all converted to grey scale, and in nii.gz format

I used MonaiLavel + Slider 3D to train how to segment a structure contained within the 2D images. I get validation dice scores north of 90%. 

I then load an image that the model didnt train with (prepared the same way, grey scale, 2D, same size, .nii.gz etc) and then run the auto segmentation feature and every pixel in the image is included as part of the segmentation mask. 

What am I doing wrong? 

I have a feeling the model the auto segmentation tool is running is not my newly trained model. Could this be the case? If so, how do I tell Slicer to now use my model. 

Thank you all! 
Mike

**Comment by tangy5** [comment]
Hi mgorin1985, yes, I feel the inference model is not the the newly trained, since the validation accuracy looks great. Not sure which model is using here, but your could see the code here (segmentation for example) how the model is loaded. 

https://github.com/Project-MONAI/MONAILabel/blob/d182253976d653537d236a0f4fb6539690b826b2/sample-apps/radiology/lib/configs/segmentation.py#L52

In addition, to check the model used for inference (auto-segmentation)., you could check the MONAI Label server logs, there should be a log for the model path. You can confirm whether the loaded model is the newly trained one.

**Comment by gorin-research-repo** [comment]
Thank you for your reply. 

I am using the segmentation.py app you have mentioned. Per a tutorial I watched online I have turned use a pretrain model to false. 

 if strtobool(self.conf.get("use_pretrained_model", "false")):

After I trained the model and achieve a performance of +80%, I have been ending the server and closing 3DSlicer and then restarting everything. This should load the newest model into the "auto segmentation feature" and the logs -- If I am reading them correctly seem to reflect this. 

I then use the feature and I get output that looks like the the following, where the entire image is just 1 big segmented rectangle. 
![image](https://user-images.githubusercontent.com/122843577/212993696-3b2e357a-f952-4d2b-be7e-6e8cee656bda.png)

At this point all I can think of is the fact that I am using a 2D image. Is it possible that the segmentation application cannot work properly on 2D data? 

Any help would be much appreciated! 

Mike

**Reply by junxiant** [reply]
Hello mgorin1985, 

were you able to solve this issue?

**Comment by SachidanandAlle** [comment]
For 2D segmentation, please refer and use endoscopy/pathology sample-apps and corresponding viewers like QuPath/DSA/CVAT
